kedro/framework/cli/starters.py

```diff
@@ -516,9 +516,10 @@ def _get_extra_context(  # noqa: PLR0913
             NUMBER_TO_TOOLS_NAME[tool]
             for tool in _parse_tools_input(tools)  # type: ignore
         ]
-        extra_context["tools"] = str(extra_context["tools"])
+    if extra_context.get("tools"):
```

Comment:
Extra condition to catch: 
- User prompts tools:"none"

Reply:
What does this additional check do on top of this check above?

```python
 tools = extra_context.get("tools")
    if tools:
```

Reply:
If the user goes through the prompts and selects none, `_parse_tools_input(tools)` on line 517 will return an empty list, which was causing the discrepancy of none resulting in either `[]` or `['None']` being returned. This additional check ensures this is handled without adding `{"none":'None'} `to `NUMBER_TO_TOOLS_NAME` which felt incongruous (as "none" isn't a number or tool selection)

Reply:
I think I understand the idea, but it seems somewhat tricky and not immediately clear, despite its functionality. Perhaps we should consider little bit refactoring lines 513-521 to clarify the logic?

Reply:
I agree, especially because this is where the three streams of input converge. I have tried to clarify in more procedural detail:

There are several ways of selecting no tools:
1. Using `--config` and omitting `tools:` from config
2. ` kedro new --tools=none`
3. User prompts and using the default value or inputing none
4. Config file entry `tools: "none"`

The condition at line 514, is triggered by cases 3 and 4, which will result in `extra_context['tools'] = []`.
Line 519 then checks against this, only converting the list to a string to be stored if the list is **not** empty. 
All cases 1-4 will skip to line 521 as `extra_context['tools']` will either be `None` or an empty list `[]`, unifying the stored result as `['None']`

I am unsure we could refactor only these lines without affecting the rest of the file, did you have any suggestions?

Reply:
@AhdraMeraliQB , I agree, it's really a lot of ways to select tools.
And I also agree that we should split some code before lines 513-521. I think we should clarify steps:
1. Finalise the value of extra_context["tools"] as a list of numbers, or [] by default anyway. It should be done before lines 513.
2. Clarify that now in lines 513 we want to convert that list of numbers to list of Readable names and [] (empty list) will be converted to a ["None"]

So split the collection and merging step from conversion step.

Reply:
I would be in favour of doing the refactoring in this PR, because the change introduced here makes the code even harder to follow. I understand why it's needed now, but it's hard to grasp it just from reading the code.